torchviz/dot.py

```diff
@@ -3,71 +3,164 @@
 from graphviz import Digraph
 import torch
 from torch.autograd import Variable
+import warnings
 
 Node = namedtuple('Node', ('name', 'inputs', 'attr', 'op'))
 
-
-def make_dot(var, params=None):
+# Saved attrs for grad_fn (incl. saved variables) begin with `._saved_*`
+SAVED_PREFIX = "_saved_"
+
+def get_fn_name(fn, show_attrs, max_attr_chars):
+    name = str(type(fn).__name__)
+    if not show_attrs:
+        return name
+    attrs = dict()
+    for attr in dir(fn):
+        if not attr.startswith(SAVED_PREFIX):
+            continue
+        val = getattr(fn, attr)
+        attr = attr[len(SAVED_PREFIX):]
+        if torch.is_tensor(val):
+            attrs[attr] = "[saved tensor]"
+        elif isinstance(val, tuple) and any(torch.is_tensor(t) for t in val):
+            attrs[attr] = "[saved tensors]"
+        else:
+            attrs[attr] = str(val)
+    if not attrs:
+        return name
+    max_attr_chars = max(max_attr_chars, 3)
+    col1width = max(len(k) for k in attrs.keys())
+    col2width = min(max(len(str(v)) for v in attrs.values()), max_attr_chars)
+    sep = "-" * max(col1width + col2width + 2, len(name))
+    attrstr = '%-' + str(col1width) + 's: %' + str(col2width)+ 's'
+    truncate = lambda s: s[:col2width - 3] + "..." if len(s) > col2width else s
+    params = '\n'.join(attrstr % (k, truncate(str(v))) for (k, v) in attrs.items())
+    return name + '\n' + sep + '\n' + params
+
+
+def make_dot(var, params=None, show_attrs=False, show_saved=False, max_attr_chars=50):
     """ Produces Graphviz representation of PyTorch autograd graph.
 
-    Blue nodes are the Variables that require grad, orange are Tensors
-    saved for backward in torch.autograd.Function
+    If a node represents a backward function, it is gray. Otherwise, the node
+    represents a tensor and is either blue, orange, or green:
+     - Blue: reachable leaf tensors that require grad
```

Comment:
nit: requires
Also you can add (Tensors whose .grad fields will be populated during .backward())

Reply:
Good idea - added!